web/src/components/torrents/TorrentCreatorDialog.tsx

```diff
@@ -0,0 +1,366 @@
+              <form.Field name="format">
+                {(field) => (
+                  <div className="space-y-2">
+                    <Label htmlFor="format">Torrent Format</Label>
+                    <Select
+                      value={field.state.value}
+                      onValueChange={(value) => field.handleChange(value as TorrentFormat | "")}
+                    >
+                      <SelectTrigger id="format">
+                        <SelectValue placeholder="Auto (v1)" />
+                      </SelectTrigger>
+                      <SelectContent>
+                        <SelectItem value="v1">v1 (Compatible)</SelectItem>
+                        <SelectItem value="v2">v2 (Modern)</SelectItem>
+                        <SelectItem value="hybrid">Hybrid (v1 + v2)</SelectItem>
+                      </SelectContent>
+                    </Select>
+                    <p className="text-sm text-muted-foreground">
+                      v1 for maximum compatibility, v2 for modern clients, hybrid for both
+                    </p>
+                  </div>
+                )}
+              </form.Field>
+
+              {/* Piece Size */}
+              <form.Field name="pieceSize">
+                {(field) => (
+                  <div className="space-y-2">
+                    <Label htmlFor="pieceSize">Piece Size (bytes)</Label>
+                    <Input
+                      id="pieceSize"
+                      type="number"
+                      placeholder="Auto (leave empty)"
+                      value={field.state.value}
+                      onChange={(e) => field.handleChange(e.target.value)}
+                    />
+                    <p className="text-sm text-muted-foreground">
+                      Leave empty for auto (recommended). Common: 16384, 32768, 65536, 131072, 262144
+                    </p>
+                  </div>
+                )}
+              </form.Field>
+
+              {/* Torrent File Path */}
+              <form.Field name="torrentFilePath">
+                {(field) => (
+                  <div className="space-y-2">
+                    <Label htmlFor="torrentFilePath">Save .torrent to (optional)</Label>
+                    <Input
+                      id="torrentFilePath"
+                      placeholder="/path/to/save/file.torrent"
+                      value={field.state.value}
+                      onChange={(e) => field.handleChange(e.target.value)}
+                    />
+                    <p className="text-sm text-muted-foreground">
+                      Where to save the .torrent file on the server
+                    </p>
+                  </div>
+                )}
+              </form.Field>
+
+              {/* URL Seeds */}
+              <form.Field name="urlSeeds">
+                {(field) => (
+                  <div className="space-y-2">
+                    <Label htmlFor="urlSeeds">Web Seeds (HTTP/HTTPS)</Label>
```

Comment:
Would be pretty fancy to have a drop down list of current announce urls in the instances.